sqlmesh/core/engine_adapter/base.py

```diff
@@ -2038,6 +2048,12 @@ def _rename_table(
     ) -> None:
         self.execute(exp.rename_table(old_table_name, new_table_name))
 
+    def _ping(self) -> None:
+        try:
+            self._execute(exp.select(exp.Literal.number(1)).sql(dialect=self.dialect))
+        finally:
+            self._connection_pool.close_cursor()
```

Comment:
Why close in all cases? If we just verified the connection is good, why not keep it?

Reply:
Is it because we are just validating the connection?

Reply:
This closing the cursor (not connection) to discard the result of the `SELECT 1` query which can technically be fetched.